File: frontend/lib/data/models/prize.dart
class Prize {
  final String prizeId;
  final String title;
  final String description;
  final int valueCents;
  final String imageUrl;
  final String sponsor;
  final int stock;
  final DateTime? createdAt;

  Prize({
    required this.prizeId,
    required this.title,
    required this.description,
    required this.valueCents,
    required this.imageUrl,
    required this.sponsor,
    required this.stock,
    this.createdAt,
  });

  factory Prize.fromJson(Map<String, dynamic> json) => Prize(
        prizeId: (json['prize_id'] ?? json['id']).toString(),
        title: (json['title'] ?? '').toString(),
        description: (json['description'] ?? '').toString(),
        valueCents: (json['value_cents'] ?? 0) is int
            ? json['value_cents'] as int
            : int.tryParse((json['value_cents'] ?? '0').toString()) ?? 0,
        imageUrl: (json['image_url'] ?? '').toString(),
        sponsor: (json['sponsor'] ?? '').toString(),
        stock: (json['stock'] ?? 0) is int
            ? json['stock'] as int
            : int.tryParse((json['stock'] ?? '0').toString()) ?? 0,
        createdAt: json['created_at'] != null
            ? DateTime.tryParse(json['created_at'].toString())
            : null,
      );

  Map<String, dynamic> toJson() => {
        'title': title,
        'description': description,
        'value_cents': valueCents,
        'image_url': imageUrl,
        'sponsor': sponsor,
        'stock': stock,
      };
}

File: backend/app/models/prize.py
import uuid
from sqlalchemy import Column, Integer, String, Text, DateTime, func, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from app.db.base import Base

class Prize(Base):
    __tablename__ = "prize"

    prize_id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    # Owner of the prize (who created it)
    user_id = Column(UUID(as_uuid=True), ForeignKey("app_user.user_id"), nullable=False)
    title = Column(String, nullable=False)
    description = Column(Text)
    value_cents = Column(Integer)
    image_url = Column(String)
    sponsor = Column(String)
    stock = Column(Integer, default=1)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
